tests/pipelines/dit/test_dit.py

```diff
@@ -154,3 +168,21 @@ def test_dit_512(self):
             )
 
             assert np.abs((expected_image - image).max()) < 1e-1
+
+    def test_dit_legacy_class_loading_from_subfolder(self):
+        ckpt_id = "facebook/DiT-XL-2-512"
+        transformer = Transformer2DModel.from_pretrained(ckpt_id, subfolder="transformer")
```

Comment:
Is there a checkpoint where the `norm_type` isn't one of `ada_norm_zero` or `ada_norm_single`? I think we should test that the `Transformer2DModel` object is returned in that case and functions normally. Perhaps another test file `test_models_transformer_2d` is needed.

Reply:
That is a different scope. Feel free to open a PR separately.

Reply:
cc @a-r-r-o-w